### PR TITLE
truncate long prompt

### PR DESCRIPTION
**Pull request by ArunSanganal**
Closes JuliaLang/julia#48115 .

### CODE REVIEW

File: src/REPLMode/REPLMode.jl [hunk 1]
```diff
@@ -518,6 +518,9 @@ function promptf()
         else
             project_name = projname(project_file)
             if project_name !== nothing
+                if length(project_name) > 20
+                    project_name = project_name[1:20] * "..."
+                end
```
Comment (IanButterworth) [inline src/REPLMode/REPLMode.jl:523]: Given the ... adds 3 chars, this should take 3 fewer chars than the check length. Also could change to a single elipses char

File: src/REPLMode/REPLMode.jl [hunk 2]
```diff
@@ -528,9 +531,11 @@ function promptf()
     if OFFLINE_MODE[]
         prefix = "$(prefix)[offline] "
     end
+
```
Comment (IanButterworth) [inline src/REPLMode/REPLMode.jl:534]:
```suggestion
```

File: src/REPLMode/REPLMode.jl [hunk 3]
```diff
@@ -528,9 +531,11 @@ function promptf()
...
     return "$(prefix)pkg> "
 end
 
+
```
Comment (IanButterworth) [inline src/REPLMode/REPLMode.jl:538]:
```suggestion
```

File: src/REPLMode/REPLMode.jl [hunk 4]
```diff
@@ -518,6 +518,9 @@ function promptf()
         else
             project_name = projname(project_file)
             if project_name !== nothing
+                if length(project_name) > 20
```
Comment (KristofferC) [inline src/REPLMode/REPLMode.jl:521]:
```suggestion
                if textwidth(project_name) > 20
```

File: src/REPLMode/REPLMode.jl [hunk 5]
```diff
@@ -518,6 +518,9 @@ function promptf()
...
+                    project_name = project_name[1:20] * "..."
```
Comment (KristofferC) [inline src/REPLMode/REPLMode.jl:522]:
```suggestion
                    project_name = first(project_name, 20) * "..."
```

**Review by IanButterworth — APPROVED** [review]
LGTM. The length can be adjusted if people find it too aggressive

**Comment by KristofferC** [comment]
I do think 20 is a bit short though.

**Comment by IanButterworth** [comment]
Truncating 1% of packages in General would be at ~29 chars

```
julia> length(Pkg.Types.Context().registries[3].pkgs)
8777

julia> for i in 20:30
       @info "$i $((count(p -> length(p[2].name) > i, Pkg.Types.Context().registries[3].pkgs) / 8777) * 100)%"
       end
[ Info: 20 8.75014241768258%
[ Info: 21 6.915802666059018%
[ Info: 22 5.38908510880711%
[ Info: 23 4.2725304773840715%
[ Info: 24 3.4522046257263304%
[ Info: 25 2.540731457217728%
[ Info: 26 1.8571265808362765%
[ Info: 27 1.560897801070981%
[ Info: 28 1.2077019482738978%
[ Info: 29 0.9342599977213172%
[ Info: 30 0.7177851202005241%
```

**Comment by IanButterworth** [comment]
> 30 should be a good number, but yes it's super easy for people to update if they feel it's too aggressive.

Sounds reasonable. Go ahead with that change and this can be merged

**Pull request merged**